app/controllers/place_controller.rb

```diff
@@ -94,7 +94,7 @@ def places_from_imminence
   end
 
   def imminence_response_from_data(postcode, imminence_response, error)
-    return ImminenceResponse.new(postcode, [], [], error) if error
+    return ImminenceResponse.new(postcode, [], [], error) if error || postcode.blank?
```

Comment:
Could we add the blank postcode logic a little higher in the chain? I think https://github.com/alphagov/frontend/blob/9b362d7879e9e239cb46b46a28ebfe770f978954/app/controllers/place_controller.rb#L72 would be a more suitable place as we're not actually handling an `imminence_response` as the validation occurs before the request.

```
return LocationError.new(INVALID_POSTCODE) if postcode.blank? || imminence_response.invalid_postcode?
```

Reply:
@1pretz1 Hmm, I've moved this up, but left it in a separate commit - the way imminence response is used throughout the controller means that a lot of places assume it exists in some form, so I'm not sure the change doesn't do more damage than the original. Have a look, see what you think.

Reply:
(It does simplify the code lower down, so it's swings and roundabouts)

Reply:
Ah yes, good point! I've had a little play based off your work and think the commit below is a reasonable solution as it doesn't really make the code more complicated as we're just moving the conditionals around and addresses the imminence response concern. Also using the `postcode_provided?` method simplifies things further: https://github.com/alphagov/frontend/commit/5b7652f66b820eb5f54c29560891b212661405e7

What do you think?

Reply:
I think I like it, works neatly for the two higher up tests. I'll give it one more once-over in the morning, but I think it covers everything. Nice work.